using-kubecost/admission-controller.md

```diff
@@ -0,0 +1,50 @@
+# Admission Controller
+
+{% hint style="info" %} The Admission Controller is a beta feature. Please read the documentation carefully. {% endhint %}
+
+Kubecost's Admission Controller is a `kubectl` integration which leverages Kubecost's Predict API to display expected future costs for changes made to CPU and RAM in your workloads. This can be used to anticipate future spend before resources have been allocated.
```

Comment:
> is a `kubectl` integration

This is not true. Kubecost's Admission Controller is a Kubernetes ["Admission Controller"](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/). The reason the demo you've probably seen uses `kubectl` is because `kubectl` is the most common/agnostic method of interacting with K8s. Admission Controllers affect all interactions with K8s, not just those which originate from `kubectl`.

The Kubecost Admission Controller currently lives within the cost-analyzer Pod today and, when enabled, hooks in to the K8s API server and instructs K8s to send all Deployment changes for review ("validation"). This behavior is captured well in the start of the following paragraph.

Non-kubectl interactions would still receive the cost prediction data and be able to display it if they wish -- it comes back as part of an "admission review". `kubectl` conveniently displays relevant parts of that review by default.

Reply:
+1 to the above from Michael. Just adding that the returned messages displayed by the API server do not even depend on using the `kubectl` program; they are returned in the payload regardless.